e2e/websocket_test.go

```diff
@@ -0,0 +1,161 @@
+package e2e
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/0xPolygon/minimal/e2e/framework"
+	"github.com/0xPolygon/minimal/jsonrpc"
+	"github.com/0xPolygon/minimal/types"
+	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/assert"
+	"github.com/umbracle/go-web3"
+)
+
+type testWSRequest struct {
+	JSONRPC string   `json:"jsonrpc"`
+	Params  []string `json:"params"`
+	Method  string   `json:"method"`
+	ID      int      `json:"id"`
+}
+
+func constructWSRequest(id int, method string, params []string) ([]byte, error) {
+	request := testWSRequest{
+		JSONRPC: "2.0",
+		Method:  method,
+		ID:      id,
+		Params:  params,
+	}
+
+	return json.Marshal(request)
+}
+
+func getWSResponse(t *testing.T, ws *websocket.Conn, request []byte) (jsonrpc.Response, error) {
```

Comment:
No need to return `error` here since you fail the test on each error in this function.

Reply:
You're right, it's redundant.
Removing now 👍